### PR TITLE
Modify dotgraph output of operations to use HTML

### LINKED ISSUE #580: Use HTML for formatting dot notes
Right now, we use [record-shape](https://graphviz.org/doc/info/shapes.html#record) to define the dot node for an [IR Operation](https://github.com/google-research/raksha/blob/e834eea668abef73bed147c7ec10e80067f45c26/src/ir/module.h#L35):

https://github.com/google-research/raksha/blob/e834eea668abef73bed147c7ec10e80067f45c26/src/ir/dot/dot_generator.cc#L139-L149

Dot supports [html-labels](https://graphviz.org/doc/info/shapes.html#html) for formatting nodes. This will be more flexible formatting and style when we want annotate nodes with results of souffle.

### PR DESCRIPTION
**Pull request by jasonwyatt**
Fixes #580

### CODE REVIEW

**Review by bgogul — COMMENTED** [review]
Looks mostly good! Thank you, Jason. Just a couple of minor code simplification suggestions. 

Were you able to test the rendering of the dot output?

File: src/ir/dot/dot_generator.cc [hunk 1]
```diff
@@ -151,44 +151,76 @@ std::string DotGeneratorHelper::GetDotNode(const Operation& op) {
...
-      op.inputs(), "|", [i = 0](std::string* out, const Value& v) mutable {
-        int node_num = i++;
-        absl::StrAppend(out,
-                        absl::StrFormat(R"(<I%d>I%d)", node_num, node_num));
-      });
   auto find_result = operation_results_.find(node_name);
   const absl::btree_set<std::string>& results =
       (find_result == operation_results_.end()) ? GetEmptyStringBtreeSet()
                                                 : find_result->second;
+  // The column_count is the LCM of the inputs and outputs, and will be used to
```
[the remark this answers is not in the document]
Comment (jasonwyatt) [reply src/ir/dot/dot_generator.cc]: Thanks! much simpler now.

File: src/ir/dot/dot_generator.cc [hunk 2]
```diff
@@ -151,44 +151,76 @@ std::string DotGeneratorHelper::GetDotNode(const Operation& op) {
...
+    input_colspan = results.size();
+    output_colspan = op.inputs().size();
+    column_count = input_colspan * output_colspan;
+  }
+
+  std::string input_ports = 
+      op.inputs().empty() 
+          ? absl::StrFormat(R"(<TD COLSPAN="%d">&nbsp;</TD>)", column_count)
+          : absl::StrJoin(
+              op.inputs(), "", [i = 0, input_colspan = input_colspan](std::string* out, const Value& v) mutable {
```
[the remark this answers is not in the document]
Comment (jasonwyatt) [reply src/ir/dot/dot_generator.cc]: Thanks again!

**Comment by jasonwyatt** [comment]
> Looks mostly good! Thank you, Jason. Just a couple of minor code simplification suggestions.
> 
> Were you able to test the rendering of the dot output?

Yeah, it renders great IMO.  [Here](https://edotor.net/?engine=dot#%23%20Place%20the%20cursor%20inside%20%22graph%22%20to%20get%20some%20refactoring%20options%0A%0Adigraph%20G%20%7B%0A%20%20node%5Bshape%3Dnone%5D%3B%0A%20%20%2F%2F%20Nodes%3A%0A%20%20B0%20%5Bshape%3DMrecord%5D%0A%20%20B0_0%20%5Blabel%3D%3C%3CTABLE%20BORDER%3D%220%22%20CELLBORDER%3D%221%22%20CELLSPACING%3D%220%22%3E%3CTR%3E%3CTD%20COLSPAN%3D%222%22%3E%26nbsp%3B%3C%2FTD%3E%3C%2FTR%3E%3CTR%3E%3CTD%20COLSPAN%3D%222%22%3Ecore.pair%3C%2FTD%3E%3C%2FTR%3E%3CTR%3E%3CTD%20COLSPAN%3D%222%22%3EnumResults(core.pair)%3A%202%20%3C%2FTD%3E%3C%2FTR%3E%3CTR%3E%3CTD%20COLSPAN%3D%221%22%20PORT%3D%22first%22%3Efirst%3C%2FTD%3E%3CTD%20COLSPAN%3D%221%22%20PORT%3D%22second%22%3Esecond%3C%2FTD%3E%3C%2FTR%3E%3C%2FTABLE%3E%3E%5D%0A%20%20B0_1%20%5Blabel%3D%3C%3CTABLE%20BORDER%3D%220%22%20CELLBORDER%3D%221%22%20CELLSPACING%3D%220%22%3E%3CTR%3E%3CTD%20COLSPAN%3D%221%22%20PORT%3D%22I0%22%3EI0%3C%2FTD%3E%3CTD%20COLSPAN%3D%221%22%20PORT%3D%22I1%22%3EI1%3C%2FTD%3E%3C%2FTR%3E%3CTR%3E%3CTD%20COLSPAN%3D%222%22%3Ecore.plus%3C%2FTD%3E%3C%2FTR%3E%3CTR%3E%3CTD%20COLSPAN%3D%222%22%3EnumResults(core.plus)%3A%200%20%3C%2FTD%3E%3C%2FTR%3E%3CTR%3E%3CTD%20COLSPAN%3D%222%22%20PORT%3D%22out%22%3Eout%3C%2FTD%3E%3C%2FTR%3E%3C%2FTABLE%3E%3E%5D%0A%20%20%2F%2F%20Edges%3A%0A%20%20B0_0%3Afirst%20-%3E%20B0_1%3AI0%0A%20%20B0_0%3Asecond%20-%3E%20B0_1%3AI1%0A%20%20%2F%2F%20End%3A%0A%7D) is [some output](https://edotor.net/?engine=dot#digraph%20G%20%7B%0A%20%20node%5Bshape%3Dnone%5D%3B%0A%20%20%2F%2F%20Nodes%3A%0A%20%20B0%20%5Bshape%3DMrecord%5D%0A%20%20B0_0%20%5Blabel%3D%3C%3CTABLE%20BORDER%3D%220%22%20CELLBORDER%3D%221%22%20CELLSPACING%3D%220%22%3E%3CTR%3E%3CTD%20COLSPAN%3D%221%22%20PORT%3D%22I0%22%3EI0%3C%2FTD%3E%3C%2FTR%3E%3CTR%3E%3CTD%20COLSPAN%3D%221%22%3Ecore.minus%3C%2FTD%3E%3C%2FTR%3E%3CTR%3E%3CTD%20COLSPAN%3D%221%22%20PORT%3D%22out%22%3Eout%3C%2FTD%3E%3C%2FTR%3E%3C%2FTABLE%3E%3E%5D%0A%20%20B0_2%20%5Blabel%3D%3C%3CTABLE%20BORDER%3D%220%22%20CELLBORDER%3D%221%22%20CELLSPACING%3D%220%22%3E%3CTR%3E%3CTD%20COLSPAN%3D%221%22%20PORT%3D%22I0%22%3EI0%3C%2FTD%3E%3CTD%20COLSPAN%3D%221%22%20PORT%3D%22I1%22%3EI1%3C%2FTD%3E%3C%2FTR%3E%3CTR%3E%3CTD%20COLSPAN%3D%222%22%3Ecore.plus%3C%2FTD%3E%3C%2FTR%3E%3CTR%3E%3CTD%20COLSPAN%3D%222%22%20PORT%3D%22out%22%3Eout%3C%2FTD%3E%3C%2FTR%3E%3C%2FTABLE%3E%3E%5D%0A%20%20B0_1%20%5Blabel%3D%3C%3CTABLE%20BORDER%3D%220%22%20CELLBORDER%3D%221%22%20CELLSPACING%3D%220%22%3E%3CTR%3E%3CTD%20COLSPAN%3D%221%22%20PORT%3D%22I0%22%3EI0%3C%2FTD%3E%3CTD%20COLSPAN%3D%221%22%20PORT%3D%22I1%22%3EI1%3C%2FTD%3E%3C%2FTR%3E%3CTR%3E%3CTD%20COLSPAN%3D%222%22%3Ecore.plus%3C%2FTD%3E%3C%2FTR%3E%3CTR%3E%3CTD%20COLSPAN%3D%222%22%20PORT%3D%22out%22%3Eout%3C%2FTD%3E%3C%2FTR%3E%3C%2FTABLE%3E%3E%5D%0A%20%20%2F%2F%20Edges%3A%0A%20%20B0_0%3Aout%20-%3E%20B0_2%3AI1%0A%20%20B0_1%3Aout%20-%3E%20B0_2%3AI0%0A%20%20B0_0%3Aout%20-%3E%20B0_1%3AI0%0A%20%20%2F%2F%20End%3A%0A%7D%0A) from the tests if you want to see what it looks like.

**Review by bgogul — APPROVED** [review]
Thank you for sharing the renderings, Jason. Looks quite good! 

LGTM. To merge this in, please add "ready to pull" label to the PR. It will kick off the copybara imports.

**Pull request closed without merging**